pkg/chat/server.go

```diff
@@ -0,0 +1,37 @@
+package chat
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+//Serve an incoming http connection
+func serveWs(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("ChatServer:$Websocket endpoint hit")
+	conn, err := Upgrade(w, r)
+	if err != nil {
+		fmt.Println("ChatServer:$Error:" + err.Error())
+		return
+	}
+	newClient := &Client{
+		ID:        strconv.Itoa(len(Clients)),
+		Conn:      conn,
+		MsgBuffer: make(chan []byte),
+	}
+
+	Register <- newClient
+	go newClient.Read()
+	go newClient.Send()
+}
+
+//Start the chat server
+func StartServer() {
+	fmt.Println("ChatServer:$System initializing")
+	go PoolLoop()
+	http.HandleFunc("/ws", func(w http.ResponseWriter, r *http.Request) {
+		serveWs(w, r)
+	})
+	fmt.Println("ChatServer:$System on")
+	http.ListenAndServe(":8080", nil)
```

Comment:
> Signed-off-by: YJ Q [57511958+Estrella142857@users.noreply.github.com](mailto:57511958+Estrella142857@users.noreply.github.com)
> 1.Remove the message in db-defi for frontend says messages will be saved locally.(as discussed for now)
> 2.Add server part for our chat module.(Don't know how it can be merged with other backend parts?)

What is defi? decentralized finance?